plugins/modules/lxd_container.py

```diff
@@ -499,6 +492,17 @@ def __init__(self, module):
             )
         except LXDClientException as e:
             self.module.fail_json(msg=e.msg)
+
+        # LXD (3.19) Rest API provides instance endpoint, failback to containers and virtual-machines
+        self.api_endpoint = '/1.0/instances'
+        try:
+            self.client.do('GET', '{0}/{1}'.format(self.api_endpoint, ''), ok_error_codes=[404])
```

Comment:
You need to check the return value whether it's a 404 error, instead of trying to catch an exception.

Reply:
When working out the best approach to do this, I found/figured out that this approach would return 404 for endpoints that didn't exist even. For example, `/1.0/test` would result in 404 like `/1.0/instances`. However, adding the `?project` filter with a blank project would result in the API response I was looking for.